internal/controller/datamovement_controller.go

```diff
@@ -516,6 +528,64 @@ func buildDMCommand(ctx context.Context, profile dmConfigProfile, hosts []string
 	return strings.Split(cmd, " "), hostfile, nil
 }
 
+func prepareDestination(dm *nnfv1alpha1.NnfDataMovement, log logr.Logger) error {
+	// TODO: verify destination is valid (i.e. global lustre or nnf directory).
+	// TODO: do the same with  the source. maybe these validations are done in sos/nnf-dm daemon.
+
+	if !isTestEnv() {
+		p, err := getDestinationDir(dm.Spec.Source.Path, dm.Spec.Destination.Path)
+		if err != nil {
+			return dwsv1alpha2.NewResourceError("could not determine source type").WithError(err).WithFatal()
+		}
+
+		if err := createDestinationDir(p, dm.Spec.UserId, dm.Spec.GroupId); err != nil {
+			return dwsv1alpha2.NewResourceError("could not create destination directory").WithError(err).WithFatal()
+		}
+		log.Info("Destination path created", "path", p)
+	}
+
+	return nil
+}
+
+// Determine the directory path to create based on the source and destination. There are 3 cases to
+// consider:
+// - directory
+// - file-file
+// - file-directory
+// If the source file doesn't exist, we cannot determine the appropriate behavior.
+func getDestinationDir(src, dest string) (string, error) {
+	// Get the source file
+	sf, err := os.Stat(src)
+	if err != nil {
+		return "", fmt.Errorf("source file does not exist: %w", err)
+	}
+	// If the source is a directory, then use the full destination path regardless of the
+	// destination type
+	if sf.IsDir() {
+		return dest, nil
+	}
+
+	// It's a file. Let Dir() determine if the destination appears to be a file or directory
+	// (trailing slash)
+	return filepath.Dir(dest), nil
+}
+
+func createDestinationDir(dest string, uid, gid uint32) error {
+	// Use exec.Command() rather than os.MkDirAll so we can set the UID/GID
+	mkdirCmd := exec.Command("mkdir", "-p", dest)
+	mkdirCmd.SysProcAttr = &syscall.SysProcAttr{
+		Credential: &syscall.Credential{
+			Uid: uid,
+			Gid: gid,
+		},
+	}
+	if err := mkdirCmd.Run(); err != nil {
```

Comment:
Try `mkdirCmd.CombinedOutput()`, so you can log everything, stdout and stderr.
https://pkg.go.dev/os/exec#Cmd.CombinedOutput